dftimewolf/lib/processors/localplaso.py

```diff
@@ -47,56 +49,73 @@ def SetUp(self, timezone: Optional[str]=None) -> None:  # pylint: disable=argume
+        self._output_path, '{0:s}.plaso'.format(uuid.uuid4().hex))
+    cmd.extend([plaso_storage_file_path, path])
+
+    # Run the l2t command
+    full_cmd = ' '.join(cmd)
+    self.logger.info('Running external command: "{0:s}"'.format(full_cmd))
+    try:
+      l2t_proc = subprocess.Popen(
+          cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+      _, error = l2t_proc.communicate()
+      l2t_status = l2t_proc.wait()
+    except OSError as exception:
+      self.ModuleError(str(exception), critical=True)
+
+    if l2t_status:
+      message = (
+          'The log2timeline command {0:s} failed: {1!s}.'
+          ' Check log file for details.').format(full_cmd, error)
+      self.ModuleError(message, critical=True)
+
+    container = containers.File(description, plaso_storage_file_path)
+    self.state.StoreContainer(container)
 
   def Process(self) -> None:
     """Executes log2timeline.py on the module input."""
+
     for file_container in self.state.GetContainers(containers.File, pop=True):
-      description = file_container.name
-      path = file_container.path
-      log_file_path = os.path.join(self._output_path, 'plaso.log')
-      self.logger.info('Log file: {0:s}'.format(log_file_path))
-
-      # Build the plaso command line.
-      cmd = [self._plaso_path]
-      # Since we might be running alongside another Module, always disable
-      # the status view.
-      cmd.extend(['-q', '--status_view', 'none'])
-      if self._timezone:
-        cmd.extend(['-z', self._timezone])
-
-      # Analyze all available partitions.
-      cmd.extend(['--partition', 'all'])
-
-      # Setup logging.
-      cmd.extend(['--logfile', log_file_path])
-
-      # And now, the crux of the command.
-      # Generate a new storage file for each plaso run
-      plaso_storage_file_path = os.path.join(
-          self._output_path, '{0:s}.plaso'.format(uuid.uuid4().hex))
-      cmd.extend([plaso_storage_file_path, path])
-
-      # Run the l2t command
-      full_cmd = ' '.join(cmd)
-      self.logger.info('Running external command: "{0:s}"'.format(full_cmd))
-      try:
-        l2t_proc = subprocess.Popen(
-            cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        _, error = l2t_proc.communicate()
-        l2t_status = l2t_proc.wait()
-      except OSError as exception:
-        self.ModuleError(str(exception), critical=True)
-
-      if l2t_status:
-        message = ('The log2timeline command {0:s} failed: {1!s}.'
-                   ' Check log file for details.').format(full_cmd, error)
-        self.ModuleError(message, critical=True)
-
-      container = containers.File(description, plaso_storage_file_path)
-      self.state.StoreContainer(container)
+      self._processContainer(file_container)
+
+    for directory_container in self.state.GetContainers(containers.Directory,
+                                                        pop=True):
+      self._processContainer(directory_container)
```

Comment:
Seeing this now, it's a little tricky. You call `GetContainers` with `pop=True`, which means they'll disappear from the store. inside `_processContainer`, you add new `File` containers. 

If these two loops had been inverted (if you had been processing `Directory` containers before the `File` containers, there would've been a bug where you would have added File containers (produced by the first loop) that would've then been re-processed.

It would be better if we popped all that needs to be popped first and stored that in a temp variable. You can specify the type of a list by appending `# type: List[Union[containers.File, containers.Directory]]` at the end of the lien where you concatenate both.